pennylane/devices/default_clifford.py

```diff
@@ -210,6 +221,50 @@ def circuit():
         Maintaining and working with this tableau representation instead of the complete state vector
         makes the calculations of increasingly large Clifford circuits more efficient on this device.
 
+    .. details::
+        :title: Probabilities for Basis States
+        :href: clifford-probabilities
+
+        One can compute the analytical probability of each computational basis state using
+        :func:`qml.probs <pennylane.probs>`, which also accepts a wire specification for obtaining marginal
+        probabilities and an observable for obtaining probabilities from the rotated computational basis.
+        As the ``default.clifford`` device supports executing quantum circuits with a large number of qubits,
+        we restrict the ability to compute the probabilities for `all` computational basis states at once to
+        maintain computational efficiency. In contrast, one can specify target state(s), i.e., subset
+        `basis states` of interest, using the ``probability_target`` property of the device.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            import numpy as np
+
+            dev = qml.device("default.clifford")
+            dev.probability_target = np.array([[0, 0], [1, 0]])
```

Comment:
A couple questions here.

* **What exactly do these numbers mean?** I'm a little confused by what the numbers actually are

* **Why can't we infer this from the measurements?** In your below example, the probability is on wires 0 and 1.  If the issue is requesting the probabilities on too many wires, can we just raise an error and say "no probabilities on 30+ different wires"?

* Why not specify on initialization? `device('default.qubit', probability_target=probability_target)`?

Reply:
+1 with @albi3ro's comment. Maybe we have a discussion of what we want to do here.

Also note that you can already use `qml.expval(qml.Projector())` for finding specific probabilities.

Reply:
> * **What exactly do these numbers mean?** I'm a little confused by what the numbers actually are

These refer to the computational basis states. For example, for a two-qubit system we would have four basis states - `|00>, |10>, |01>, |11>` which would correspond to `[0, 0], [1, 0], [0, 1], [1, 1]`.

> * **Why can't we infer this from the measurements?** In your below example, the probability is on wires 0 and 1.  If the issue is requesting the probabilities on too many wires, can we just raise an error and say "no probabilities on 30+ different wires"?

I did think about it. Added support for it now and restricted qubits size to `24`. 

> * Why not specify on initialization? `device('default.qubit', probability_target=probability_target)`?

Added a keyword argument for specifying it during initialization.